presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -0,0 +1,428 @@
+      auto split =
+          exec::Split(std::make_shared<exec::RemoteConnectorSplit>(taskId), -1);
+      task->addSplit("0", std::move(split));
+    }
+    task->noMoreSplits("0");
+  }
+
+  RowVectorPtr deserialize(
+      const RowVectorPtr& serializedResult,
+      const RowTypePtr& rowType) {
+    auto serializedData =
+        serializedResult->childAt(1)->as<FlatVector<StringView>>();
+
+    // Serialize data into a single block.
+
+    // Calculate total size.
+    size_t totalSize = 0;
+    for (auto i = 0; i < serializedData->size(); ++i) {
+      totalSize += serializedData->valueAt(i).size();
+    }
+
+    // Allocate the block. Add an extra sizeof(size_t) bytes for each row to
+    // hold row size.
+    BufferPtr buffer = AlignedBuffer::allocate<char>(
+        totalSize + sizeof(size_t) * serializedData->size(), pool());
+    auto rawBuffer = buffer->asMutable<char>();
+
+    // Copy data.
+    size_t offset = 0;
+    for (auto i = 0; i < serializedData->size(); ++i) {
+      auto value = serializedData->valueAt(i);
+
+      *(size_t*)(rawBuffer + offset) = value.size();
+      offset += sizeof(size_t);
+
+      memcpy(rawBuffer + offset, value.data(), value.size());
+      offset += value.size();
+    }
+
+    // Deserialize the block.
+    return deserialize(buffer, rowType);
+  }
+
+  RowVectorPtr deserialize(BufferPtr& serialized, const RowTypePtr& rowType) {
+    auto serializer =
+        std::make_unique<serializer::spark::UnsafeRowVectorSerde>();
+
+    ByteRange byteRange = {
+        serialized->asMutable<uint8_t>(), (int32_t)serialized->size(), 0};
+
+    auto input = std::make_unique<ByteStream>();
+    input->resetInput({byteRange});
+
+    RowVectorPtr result;
+    serializer->deserialize(input.get(), pool(), rowType, &result, nullptr);
+    return result;
+  }
+
+  RowVectorPtr rowVector(const std::vector<VectorPtr>& children) {
+    std::vector<std::string> names;
+    for (int32_t i = 0; i < children.size(); ++i) {
+      names.push_back(fmt::format("c{}", i));
+    }
+
+    return rowVector(std::move(names), children);
+  }
+
+  RowVectorPtr rowVector(
+      std::vector<std::string> childNames,
+      const std::vector<VectorPtr>& children) {
+    std::vector<std::shared_ptr<const Type>> childTypes;
+    childTypes.resize(children.size());
+    for (int i = 0; i < children.size(); i++) {
+      childTypes[i] = children[i]->type();
+    }
+    auto rowType = ROW(std::move(childNames), std::move(childTypes));
+    const size_t vectorSize = children.empty() ? 0 : children.front()->size();
+
+    return std::make_shared<RowVector>(
+        pool_.get(), rowType, BufferPtr(nullptr), vectorSize, children);
+  }
+
+  RowVectorPtr rowVector(
+      const std::shared_ptr<const RowType>& rowType,
+      vector_size_t size) {
+    return std::make_shared<RowVector>(
+        pool_.get(),
+        rowType,
+        BufferPtr(nullptr),
+        size,
+        std::vector<VectorPtr>{});
+  }
+
+  RowVectorPtr makeRowVector(
```

Comment:
This method should not be needed. We should use the implementation from the base class. Then, 'rowVector' method won't be needed as well.

Reply:
Same.